src/peergos/server/util/DifficultyGenerator.java

```diff
@@ -0,0 +1,56 @@
+package peergos.server.util;
+
+import peergos.shared.crypto.*;
+
+public class DifficultyGenerator {
+
+    private final RateMonitor queryRate;
+    private int difficulty = ProofOfWork.MIN_DIFFICULTY;
+    private long timeOfLastUpdateMillis;
+    private final long[] maxPerBucket;
+
+    public DifficultyGenerator(long startTimeMillis, int maxPerDay) {
```

Comment:
Please be kind and  add a comment describing  what this is supposed  to  do.

Reply:
done